streaming/vision/ade20k.py

```diff
@@ -0,0 +1,126 @@
+        if min_resize_scale <= 0:
+            raise ValueError('min_resize_scale must be positive')
+        if max_resize_scale <= 0:
+            raise ValueError('max_resize_scale must be positive')
+        if max_resize_scale < min_resize_scale:
+            raise ValueError('max_resize_scale cannot be less than min_resize_scale')
+        if final_size <= 0:
+            raise ValueError('final_size must be positive')
+
+        super().__init__(local, split, remote, shuffle, transforms, transform, target_transform,
+                         prefetch, keep_zip, retry, timeout, hash, batch_size)
+
+        r_mean, g_mean, b_mean = IMAGENET_CHANNEL_MEAN
+
+        if split == 'train':
+            if not self.transform:
+                self.transform = torch.nn.Sequential(
+                    RandomResizePair(min_scale=min_resize_scale,
+                                     max_scale=max_resize_scale,
+                                     base_size=(base_size, base_size)),
+                    RandomCropPair(
+                        crop_size=(final_size, final_size),
+                        class_max_percent=0.75,
+                        num_retry=10,
+                    ),
+                    RandomHFlipPair(),
+                    # Photometric distoration values come from mmsegmentation:
+                    # https://github.com/open-mmlab/mmsegmentation/blob/aa50358c71fe9c4cccdd2abe42433bdf702e757b/mmseg/datasets/pipelines/transforms.py#L861
+                    PhotometricDistoration(brightness=32. / 255,
+                                           contrast=0.5,
+                                           saturation=0.5,
+                                           hue=18. / 255),
+                    PadToSize(size=(final_size, final_size),
+                              fill=(int(r_mean), int(g_mean), int(b_mean))))
+            if not self.target_transform:
+                self.target_transform = torch.nn.Sequential(
+                    RandomResizePair(min_scale=min_resize_scale,
+                                     max_scale=max_resize_scale,
+                                     base_size=(base_size, base_size)),
+                    RandomCropPair(
+                        crop_size=(final_size, final_size),
+                        class_max_percent=0.75,
+                        num_retry=10,
+                    ), RandomHFlipPair(), PadToSize(size=(final_size, final_size), fill=0))
+        else:
+            if not self.transform:
+                self.transform = T.Resize(size=(final_size, final_size),
+                                          interpolation=TF.InterpolationMode.BILINEAR)
+            if not self.target_transform:
+                self.target_transform = T.Resize(size=(final_size, final_size),
+                                                 interpolation=TF.InterpolationMode.NEAREST)
+
+    def __getitem__(self, idx: int) -> Tuple[Any, Any]:
```

Comment:
should we say Tuple[Tensor, Tensor] here?

Reply:
Yes. Done. Thanks!

Reply:
I moved back to `Tuple[Any, Any]` for now in https://github.com/mosaicml/streaming/pull/9/commits/6972e99881cba28a7035b300973d58e07a5b0509 since if all the transforms are None then it would return as PIL Image.

Reply:
D'oh